### PR TITLE
feat: syntax highlighting using chroma

### PR DESCRIPTION
**Pull request by elianiva**
image

### CODE REVIEW

File: platform/highlight/highlight.go [hunk 1]
```diff
@@ -0,0 +1,47 @@
...
+	style := styles.Get(theme)
+	if style == nil {
+		// default theme is dracula
+		style = styles.Dracula
+	}
+
+	iterator, err := lexer.Tokenise(nil, source)
+	if err != nil {
+		return source
+	}
```
Comment (aldy505) [inline platform/highlight/highlight.go:31]: error should be explicitly returned, change the function output signature to `(string, error)` please

File: platform/highlight/highlight.go [hunk 2]
```diff
@@ -0,0 +1,47 @@
...
+	if linenr != "" {
+		opts = append(opts, html.WithLineNumbers(true))
+	}
+
+	var buf bytes.Buffer
+	formatter := html.New(opts...)
+	err = formatter.Format(&buf, style, iterator)
+	if err != nil {
+		return source
+	}
```
Comment (aldy505) [inline platform/highlight/highlight.go:44]: ditto

File: handlers/get.go [hunk 3]
```diff
@@ -53,6 +55,16 @@ func (d *Dependency) Get(c *fiber.Ctx) error {
 		}
 	}
 
+	// we need to replace escaped newline to literal newline
+	content := strings.Replace(i.Paste, `\n`, "\n", -1)
+
+	if qs.Language != "" {
+		highlighted := h.Highlight(content, qs.Language, qs.Theme, qs.LineNr)
```
Comment (aldy505) [inline handlers/get.go:62]: see my notes on `highlight.go`

**Review by aldy505 — CHANGES_REQUESTED** [review]

File: handlers/get.go [hunk 4]
```diff
@@ -58,13 +58,18 @@ func (d *Dependency) Get(c *fiber.Ctx) error {
 	// we need to replace escaped newline to literal newline
 	content := strings.Replace(i.Paste, `\n`, "\n", -1)
 
+	c.Set("Content-Type", "text/plain")
 	if qs.Language != "" {
-		highlighted := h.Highlight(content, qs.Language, qs.Theme, qs.LineNr)
+		highlighted, err := h.Highlight(content, qs.Language, qs.Theme, qs.LineNr)
+		if err != nil {
+			// they should still be able to get the plain text even if the highlighter is b0rked
+			c.Status(http.StatusOK).Send([]byte(content))
```
Comment (aldy505) [inline handlers/get.go:66]: if it's error, just `return err`, no need to make a `StatusOK` response

File: handlers/get.go [hunk 5]
```diff
@@ -58,13 +58,18 @@ func (d *Dependency) Get(c *fiber.Ctx) error {
 	// we need to replace escaped newline to literal newline
 	content := strings.Replace(i.Paste, `\n`, "\n", -1)
 
+	c.Set("Content-Type", "text/plain")
```
Comment (aldy505) [inline handlers/get.go:61]: the `Content-Type` headers will be set twice when `qs.Language` is not empty if you put it here

**Review by aldy505 — CHANGES_REQUESTED** [review]
little bit more

File: handlers/get.go (hunk 4 above)
Comment (elianiva) [reply handlers/get.go:66]: kalo highlighternya doang yg error, masa ga ditampilin meskipun ga pake highlight?

File: handlers/get.go (hunk 5 above)
Comment (elianiva) [reply handlers/get.go:61]: alasan kenapa ditaruh disini relate sama ini https://github.com/teknologi-umum/polarite/pull/1#discussion_r722002352

File: handlers/get.go (hunk 4 above)
Comment (aldy505) [reply handlers/get.go:66]: kemungkinan highlighternya error gara-gara apa?

Comment (elianiva) [reply handlers/get.go:66]: gagal tokenise atau formatter gagal ngehasilin output

Comment (aldy505) [reply handlers/get.go:66]: oke. tetep aja ga ideal. tapi aku benerin nantian

**Pull request merged**